controllers/drplacementcontrol_controller.go

```diff
@@ -364,18 +365,32 @@ func (r *DRPlacementControlReconciler) Reconcile(ctx context.Context, req ctrl.R
 }
 
 func (r *DRPlacementControlReconciler) recordFailure(drpc *rmn.DRPlacementControl,
-	usrPlRule *plrv1.PlacementRule, reason, msg string, log logr.Logger,
+	usrPlRule *plrv1.PlacementRule, reason, msg string, syncMetrics *SyncMetrics, log logr.Logger,
 ) {
 	needsUpdate := SetDRPCStatusCondition(&drpc.Status.Conditions, rmn.ConditionAvailable,
 		drpc.Generation, metav1.ConditionFalse, reason, msg)
 	if needsUpdate {
-		err := r.updateDRPCStatus(drpc, usrPlRule, log)
+		err := r.updateDRPCStatus(drpc, usrPlRule, syncMetrics, log)
 		if err != nil {
 			log.Info(fmt.Sprintf("Failed to update DRPC status (%v)", err))
 		}
 	}
 }
 
+func (r *DRPlacementControlReconciler) setLastSyncTimeMetric(syncMetrics *SyncMetrics,
+	t *metav1.Time, log logr.Logger,
+) {
+	if syncMetrics != nil {
+		log.Info(fmt.Sprintf("Setting metric: (%s)", LastSyncTimestampSeconds))
+
+		if t == nil {
+			syncMetrics.LastSyncTime.Set(0)
```

Comment:
There should be a return after this?

Reply:
`set` does not return anything. so unless we send `err` or something, i don't think there is anything to return from this func.

Reply:
If we do not add an empty return, then code will flow to line 390 and access `t` which is nil. As we have set the metric, we should return right away within this if block to prevent the same.

Also, from a style perspective, the following maybe a good refactor here.:
```
if syncMetrics == nil {
  return
}

if t == nil {
  syncMetrics.LastSyncTime.Set(0)
  return
}

syncMetrics.LastSyncTime.Set(float64(t.ProtoTime().Seconds))
```

Reply:
Resolved in this [commit](https://github.com/RamenDR/ramen/pull/722/commits/0defd924c6857338c24d39b74d850de9cf1e3ca5)